semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py

```diff
@@ -271,8 +271,12 @@ class AttributeType:
 
     @staticmethod
     def get_type(t):
-        if isinstance(t, numbers.Number):
+        if isinstance(t, int):
             return "int"
+        if isinstance(t, numbers.Number):
+            raise ValidationError(
```

Comment:
Shouldn't this be validated "eagerly"? I.e., even if get_type is not called on the enum, there should be an error (maybe get_type is currently always called, but logically it could make more sense to have this somewhere else).

Reply:
`get_type` is called only from `EnumAttributeType.parse`, so I considered this part of enum parsing but I can move the validation directly into `EnumAttributeType.parse` to make it cleaner.

Reply:
In that case, just renaming the function or maybe inlining it might be more clear.

Reply:
Just noticed it's also used in one other place, so I left `get_type` itself as such.